pkg-manager/modules-cleaner/src/removeDirectDependency.ts

```diff
@@ -42,8 +42,11 @@ export async function removeDirectDependency (
 }
 
 export async function removeIfEmpty (dir: string) {
-  if (await dirIsEmpty(dir)) {
-    await rimraf(dir)
+  try {
+    if (await dirIsEmpty(dir)) {
+      await rimraf(dir)
```

Comment:
only this part fails, no? So I guess put only rimraf to try/catch. And add a comment about in what scenario it might happen.

Reply:
![image](https://github.com/pnpm/pnpm/assets/23251408/947202ab-6c4d-498a-ba49-52e8bd04bc7b)
This causes pnpm install to fail, causing the build process to terminate with an error.